cpp/include/cudf_test/iterator_utilities.hpp

```diff
@@ -89,32 +87,49 @@ template <typename Iter>
  * and yields `true` (to mark valid rows) for all other indices. E.g.
  *
  * @code
- * auto iter = iterator_with_null_at(8);
+ * auto iter = null_at(8);
  * iter[7] == true;  // i.e. Valid row at index 7.
  * iter[8] == false; // i.e. Invalid row at index 8.
  * @endcode
  *
  * @param index The index for which the validity iterator must return `false` (i.e. null)
  * @return auto Validity iterator
  */
-[[maybe_unused]] static auto iterator_with_null_at(cudf::size_type index)
+[[maybe_unused]] static auto null_at(cudf::size_type index)
 {
-  return iterator_with_null_at(std::vector<size_type>{index});
+  return null_at(std::vector<cudf::size_type>{index});
 }
 
 /**
  * @brief Bool iterator for marking all elements are null
  *
  * @return auto Validity iterator which always yields `false`
  */
-[[maybe_unused]] static auto iterator_all_nulls() { return thrust::make_constant_iterator(false); }
+[[maybe_unused]] static auto all_nulls() { return thrust::make_constant_iterator(false); }
```

Comment:
The singular/plural usage in all of these null iterators is inconsistent:
 * `null_at` -- there are versions of this that generate multiple nulls, and one version that generates a single null.
 * `all_nulls` -- generates only nulls
 * `no_null` -- generates no nulls.
 * `nulls_from_nullptr` -- why is `nulls` plural and `nullptr` is not?

I think we should be consistent -- make these all plural: `nulls_at`, `all_nulls`, `no_nulls`, `nulls_from_nullptrs`.